cuML/CMakeLists.txt

```diff
@@ -31,6 +31,12 @@ else()
   message(STATUS "Manually setting BLAS to ${BLAS_LIBRARIES}")
 endif()
 
+if(NOT DEFINED LAPACK_LIBRARIES)
+  find_package( LAPACK REQUIRED )
+else()
+  message(STATUS "Manually setting LAPACK to ${LAPACK_LIBRARIES}")
+endif()
```

Comment:
I don't like this approach, because if `LAPACK_LIBRARIES` isn't set and CMake doesn't find `LAPACK` then you mysteriously fail in the faiss configuration as opposed to CMake yelling loud and clear.

Reply:
Doesn't the `REQUIRED` in` find_package( LAPACK REQUIRED )` make it so that if `LAPACK_LIBRARIES` is not setup, it requires to find `LAPACK`?

Reply:
While definitely important that we allow the users to override the BLAS_LIBRARIES and LAPACK_LIBRARIES options in cuml, why are we then hardcoding them in FAISS? 

I don’t like the idea of assuming that users have conda in order to build the c++ library. I will be more comfortable using a hard coded lib path only after we have verified a user has not provided one explicitly and we have unsuccessfully found the lib on the system.

Reply:
(Deleted my comment. I’m on my phone and did not realize I was looking at tthe build.sh)

Reply:
> Doesn't the `REQUIRED` in` find_package( LAPACK REQUIRED )` make it so that if `LAPACK_LIBRARIES` is not setup, it requires to find `LAPACK`?

Yes, sorry. Would still prefer to use standard CMake fixtures of `LAPACK_ROOT` instead of manually setting `LAPACK_LIBRARIES`